src/hypervisor-integration.adoc

```diff
@@ -0,0 +1,199 @@
+
+The <<vstvecc>> register is a renamed extension of <<vstvec>> that is able to
+hold a capability. Its reset value is the <<infinite-cap>> capability.
+
+.Virtual supervisor trap vector base address capability register
+include::img/vstveccreg.edn[]
+
+The handling of <<vstvecc>> is otherwise identical to <<mtvecc>>, but in
+virtual supervisor mode.
+
+[#vsscratch,reftext="vsscratch"]
+=== Virtual Supervisor Scratch Register (vsscratch)
+
+The <<vsscratch>> register is as defined in cite:[riscv-priv-spec]. It is a
+VSXLEN read/write register that is VS-mode's version of supervisor register
+<<sscratch>>. <<vsscratch>> is extended into <<vsscratchc>>.
+
+.Virtual supervisor scratch register
+include::img/vsscratchreg.edn[]
+
+[#vsscratchc,reftext="vsscratchc"]
+=== Virtual Supervisor Scratch Register (vsscratchc)
+
+The <<vsscratchc>> register is a renamed version of <<vsscratch>> that is able
+to hold a capability.
+
+{TAG_RESET_CSR}
+
+It is not WARL, all capability fields must be implemented.
+
+.Virtual supervisor scratch capability register
+include::img/vsscratchcreg.edn[]
+
+[#vsepc,reftext="vsepc"]
+=== Virtual Supervisor Exception Program Counter (vsepc)
+
+The <<vsepc>> register is as defined in cite:[riscv-priv-spec]. It is extended
+into <<vsepcc>>.
+
+.Virtual supervisor exception program counter
+include::img/vsepcreg.edn[]
+
+[#vsepcc,reftext="vsepcc"]
+=== Virtual Supervisor Exception Program Counter Capability (vsepcc)
+
+The <<vsepcc>> register is a renamed extension of <<vsepc>> that is able to
+hold a capability. Its reset value is the <<infinite-cap>> capability.
+
+As shown in xref:CSR_exevectors[xrefstyle=short], <<vsepcc>> is an executable
+vector, so it need not be able to hold all possible invalid addresses.
+Additionally, the capability in <<vsepcc>> is unsealed when it is installed in
+<<pcc>> on execute of an <<SRET>> instruction. The handling of <<vsepcc>> is
+otherwise identical to <<mepcc>>, but in virtual supervisor mode.
+
+.Virtual supervisor exception program counter capability
+include::img/vsepccreg.edn[]
+
+[#vscause,reftext="vscause"]
+=== Virtual Supervisor Cause Register (vscause)
+
+The <<vscause>> register is as defined in cite:[riscv-priv-spec]. It must
+additionally support the new exception code for CHERI exceptions that
+<<scause>> supports.
+
+[#vstdc,reftext="vstdc"]
+=== Virtual Supervisor Trap Default Capability Register (vstdc)
+
+The <<vstdc>> register is a capability width read/write register that is
+VS-mode's version of supervisor register <<stdc>>. This register is only
+present when the implementation supports {cheri_default_ext_name}.
+
+{TAG_RESET_CSR}
+
+{REQUIRE_CRE_CSR}
+
+.Virtual supervisor trap default capability register
+include::img/vstdcreg.edn[]
+
+=== Existing Hypervisor Load and Store Instructions
+
+The hypervisor extension defines several integer load and store instruction
+(HLV.*x*, HSV.*x* and HLVX.*x*) that transfer the amount of integer data
```

Comment:
it would be better to have xrefs to the instructions here, although there are several, but to make it clear that the instruction pages do exist

Reply:
Fixed